src/java.base/share/classes/java/lang/Thread.java

```diff
@@ -782,29 +772,22 @@ public Thread(ThreadGroup group, Runnable target, String name,
      * @see        #stop()
      */
     public synchronized void start() {
-        /**
-         * This method is not invoked for the main method thread or "system"
-         * group threads created/set up by the VM. Any new functionality added
-         * to this method in the future may have to also be added to the VM.
-         *
-         * A zero status value corresponds to state "NEW".
-         */
-        if (threadStatus != 0)
-            throw new IllegalThreadStateException();
-
-        /* Notify the group that this thread is about to be started
-         * so that it can be added to the group's list of threads
-         * and the group's unstarted count can be decremented. */
-        group.add(this);
+        boolean success = false;
 
-        boolean started = false;
         try {
-            start0();
-            started = true;
+            synchronized (blockerLock) {
+                if (started) {
+                    throw new IllegalThreadStateException();
+                }
+
+                group.add(this);
```

Comment:
The call to `ThreadGroup.add()` should not be within the `try` block; `ThreadGroup.threadStartFailed()` should not be called if `add()` did not succeed.

Reply:
This is existing OpenJ9 behaviours, a change moving `group.add(this)` out of `try` block would be accepted in `OpenJ9` head stream first.

Reply:
The list of fixes that would be deferred is getting a little too large for my taste. I take back my agreement to defer the known problems discussed here.

Reply:
Just going back a bit about this PR, it is to adopt RI `j.l.Thread` and support on-going development of project Loom, the change requires `--enable-openjdk-thread-support`, and not enabled by default.
This PR intended to keep existing OpenJ9 behaviours while consuming additional Loom related APIs w/o re-implementation in OpenJ9.
If there is a concern about existing OpenJ9 behaviours, I would like it is proposed in OpenJ9 head stream, and go through usual tests before porting here which doesn't have similar test coverage yet.
Does this make sense?

Reply:
Ported https://github.com/eclipse-openj9/openj9/pull/14355 here, moved `started` checking & `group.add(this)` out of the `try` block, also removed the check of `started` at the `finally` to match RI behaviours.